localization/pose_initializer/src/pose_initializer/pose_initializer_core.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright 2020 Autoware Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "pose_initializer_core.hpp"
+
+#include "copy_vector_to_array.hpp"
+#include "gnss_module.hpp"
+#include "ndt_module.hpp"
+#include "stop_check_module.hpp"
+
+#include <memory>
+#include <vector>
+
+PoseInitializer::PoseInitializer() : Node("pose_initializer")
+{
+  const auto node = component_interface_utils::NodeAdaptor(this);
+  group_srv_ = create_callback_group(rclcpp::CallbackGroupType::MutuallyExclusive);
+  node.init_pub(pub_state_);
```

Comment:
Sorry if I misunderstood. 
It seems that `pub_state_` don't have a transient local option.
And it seems that it is only published when the state is changed.
Depending on the startup timing of the node subscribing this topic, it may not receive the topic until the state changes, so it is better to add the Transient local option.

Reply:
Sorry. I missed it.
https://github.com/autowarefoundation/autoware.universe/blob/d057b4afa1df71fee2b019af10e7cbf16bc9a9e6/common/component_interface_specs/include/component_interface_specs/localization.hpp#L38